src/foam/dao/RestDAO.js

```diff
@@ -0,0 +1,169 @@
+  /**
+ * @license
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+foam.CLASS({
+  package: 'foam.dao',
+  name: 'RestDAO',
+  extends: 'foam.dao.AbstractDAO',
+
+  documentation:
+  function() {/*
+               A client-side DAO for interacting with a REST endpoint.
+
+               Sinks are managed on the client (i.e., sinks passed to
+               select() will not serialize the sink and send it to the
+               endpoint for server-side logic implementation).
+               */},
+
+  requires: [
+    'foam.dao.ArraySink',
+    'foam.net.HTTPRequest'
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'baseURL',
+      documentation: 'URL for most rest calls. Some calls add "/<some-info>".',
+      final: true,
+      factory: function() {
```

Comment:
No such initialization phase. A `RestDAO` with no `baseURL` "has no meaning". If you don't know the `baseURL` yet, you're not ready to construct a `RestDAO` yet.